EDI parsing/Unparsing fails in case of nested jars

Because of the following bug introduced in daffodil v3.6.0
https://issues.apache.org/jira/browse/DAFFODIL-2892

Smooks version RC3 and above fail to parse and unparse EDI.

Steps to reproduce:
1. Create a jar with EDI parsing or unparsing example.
2. deploy that jar and try to invoke EDI parsing/unparsing.
3. The transfprmation fails with following stacktrace: 

Caused by: org.smooks.api.SmooksConfigException: org.smooks.api.SmooksConfigException: java.lang.RuntimeException: org.apache.daffodil.lib.exceptions.Abort: Invariant broken: parts.length.==(2)
org.apache.daffodil.lib.exceptions.Assert$.abort(Assert.scala:159)
org.apache.daffodil.lib.util.Misc$.optRelativeJarFileURI(Misc.scala:183)
org.apache.daffodil.lib.util.Misc$.getResourceRelativeOnlyOption(Misc.scala:145)
org.apache.daffodil.lib.xml.XMLUtils$.$anonfun$resolveSchemaLocation$3(XMLUtils.scala:1492)
	at org.smooks.cartridges.dfdl.unparser.DfdlUnparserContentHandlerFactory.create(DfdlUnparserContentHandlerFactory.java:74)
	at org.smooks.cartridges.dfdl.unparser.DfdlUnparserContentHandlerFactory.create(DfdlUnparserContentHandlerFactory.java:56)
	at org.smooks.engine.delivery.DefaultContentDeliveryConfigBuilder$ContentHandlerExtractionStrategy.addContentDeliveryUnit(DefaultContentDeliveryConfigBuilder.java:442)
	at org.smooks.engine.delivery.DefaultContentDeliveryConfigBuilder$ContentHandlerExtractionStrategy.applyContentDeliveryUnitStrategy(DefaultContentDeliveryConfigBuilder.java:399)
	at org.smooks.engine.delivery.DefaultContentDeliveryConfigBuilder$ContentHandlerExtractionStrategy.applyStrategy(DefaultContentDeliveryConfigBuilder.java:373)
	at org.smooks.engine.delivery.DefaultContentDeliveryConfigBuilder$ResourceConfigTableIterator.iterate(DefaultContentDeliveryConfigBuilder.java:524)
	at org.smooks.engine.delivery.DefaultContentDeliveryConfigBuilder$ResourceConfigTableIterator.access$200(DefaultContentDeliveryConfigBuilder.java:504)
	at org.smooks.engine.delivery.DefaultContentDeliveryConfigBuilder.extractContentHandlers(DefaultContentDeliveryConfigBuilder.java:346)
	at org.smooks.engine.delivery.DefaultContentDeliveryConfigBuilder.load(DefaultContentDeliveryConfigBuilder.java:238)
	at org.smooks.engine.delivery.DefaultContentDeliveryConfigBuilder.build(DefaultContentDeliveryConfigBuilder.java:140)
	at org.smooks.engine.delivery.DefaultContentDeliveryRuntimeFactory.create(DefaultContentDeliveryRuntimeFactory.java:86)
	at org.smooks.engine.DefaultExecutionContext.<init>(DefaultExecutionContext.java:117)
	at org.smooks.engine.DefaultExecutionContext.<init>(DefaultExecutionContext.java:94)
	at org.smooks.Smooks.createExecutionContext(Smooks.java:438)
	at org.smooks.Smooks.createExecutionContext(Smooks.java:404)
	at org.smooks.engine.resource.visitor.smooks.NestedSmooksVisitor.filterSource(NestedSmooksVisitor.java:398)
	at org.smooks.engine.resource.visitor.smooks.NestedSmooksVisitor.replaceBefore(NestedSmooksVisitor.java:325)
	at org.smooks.engine.resource.visitor.smooks.NestedSmooksVisitor.visitBefore(NestedSmooksVisitor.java:264)
	at org.smooks.engine.delivery.interceptor.AbstractInterceptorVisitor$VisitBeforeInvocation.invoke(AbstractInterceptorVisitor.java:178)
	at org.smooks.engine.delivery.interceptor.AbstractInterceptorVisitor$VisitBeforeInvocation.invoke(AbstractInterceptorVisitor.java:175)
	at org.smooks.engine.delivery.interceptor.AbstractInterceptorVisitor.intercept(AbstractInterceptorVisitor.java:122)
	at org.smooks.engine.delivery.interceptor.EventInterceptor.visitBefore(EventInterceptor.java:67)
	at org.smooks.engine.delivery.interceptor.AbstractInterceptorVisitor$VisitBeforeInvocation.invoke(AbstractInterceptorVisitor.java:178)
	at org.smooks.engine.delivery.interceptor.AbstractInterceptorVisitor$VisitBeforeInvocation.invoke(AbstractInterceptorVisitor.java:175)
	at org.smooks.engine.delivery.interceptor.AbstractInterceptorVisitor.intercept(AbstractInterceptorVisitor.java:122)
	at org.smooks.engine.delivery.interceptor.TextConsumerInterceptor.visitBefore(TextConsumerInterceptor.java:89)
	at org.smooks.engine.delivery.interceptor.AbstractInterceptorVisitor$VisitBeforeInvocation.invoke(AbstractInterceptorVisitor.java:178)
	at org.smooks.engine.delivery.interceptor.AbstractInterceptorVisitor$VisitBeforeInvocation.invoke(AbstractInterceptorVisitor.java:175)
	at org.smooks.engine.delivery.interceptor.AbstractInterceptorVisitor.intercept(AbstractInterceptorVisitor.java:122)
	at org.smooks.engine.delivery.interceptor.StreamResultWriterInterceptor.visitBefore(StreamResultWriterInterceptor.java:87)
	at org.smooks.engine.delivery.interceptor.AbstractInterceptorVisitor$VisitBeforeInvocation.invoke(AbstractInterceptorVisitor.java:178)
	at org.smooks.engine.delivery.interceptor.AbstractInterceptorVisitor$VisitBeforeInvocation.invoke(AbstractInterceptorVisitor.java:175)
	at org.smooks.engine.delivery.interceptor.AbstractInterceptorVisitor.intercept(AbstractInterceptorVisitor.java:122)
	at org.smooks.engine.delivery.interceptor.ExceptionInterceptor.intercept(ExceptionInterceptor.java:106)
	at org.smooks.engine.delivery.interceptor.ExceptionInterceptor.visitBefore(ExceptionInterceptor.java:86)
	at org.smooks.engine.delivery.interceptor.AbstractInterceptorVisitor$VisitBeforeInvocation.invoke(AbstractInterceptorVisitor.java:178)
	at org.smooks.engine.delivery.interceptor.AbstractInterceptorVisitor$VisitBeforeInvocation.invoke(AbstractInterceptorVisitor.java:175)
	at org.smooks.engine.delivery.interceptor.AbstractInterceptorVisitor.intercept(AbstractInterceptorVisitor.java:122)
	at org.smooks.engine.delivery.interceptor.StaticProxyInterceptor.visitBefore(StaticProxyInterceptor.java:120)
	at org.smooks.engine.delivery.sax.ng.SaxNgHandler.visitBefore(SaxNgHandler.java:284)
	at org.smooks.engine.delivery.sax.ng.SaxNgHandler.startElement(SaxNgHandler.java:186)
	at org.smooks.engine.delivery.SmooksContentHandler.startElement(SmooksContentHandler.java:110)
	at com.ctc.wstx.sr.BasicStreamReader.fireSaxStartElement(BasicStreamReader.java:1787)
	at com.ctc.wstx.sax.WstxSAXParser.fireStartTag(WstxSAXParser.java:806)
	at com.ctc.wstx.sax.WstxSAXParser.fireEvents(WstxSAXParser.java:691)
	at com.ctc.wstx.sax.WstxSAXParser.parse(WstxSAXParser.java:623)
	at org.smooks.engine.delivery.sax.ng.SaxNgParser.parse(SaxNgParser.java:86)
	at org.smooks.engine.delivery.sax.ng.SaxNgFilter.doFilter(SaxNgFilter.java:110)
	... 158 more
<!--
You should first reach out through the Smooks user mailing list (https://groups.google.com/g/smooks-user) where the community is more active. Alternatively, take time to read FAQs (https://www.smooks.org/faq.html) and the user guide (https://www.smooks.org/documentation/), or search the mailing list archives to find answers. Create a bug report only when you are positive that the issue is indeed a bug.

Do NOT report Security Vulnerabilities here. Please open a draft security advisory at https://github.com/smooks/[repository]/security/advisories/new where [repository] is substituted with this repository's name.
-->

**Describe the bug**

A clear and concise description of what the bug is.

**To Reproduce**

Steps to reproduce the behavior.

**Expected behavior**

A clear and concise description of what you expected to happen.

**Environment**

<!--
Example:
- Smooks: 2.0.0-RC1
- Java: OpenJDK 8
- OS: Ubuntu 20.04
-->

**Sample**

A link to a GitHub repository with a [minimal, reproducible sample](https://stackoverflow.com/help/minimal-reproducible-example).

Reports that include a sample will take priority over reports that do not. At times, we may require a sample, so it is good to try and include a sample up front.

--- a/common-schemas/src/main/resources/EDIFACT-Common/IBM_EDI_Format.dfdl.xsd
+++ b/common-schemas/src/main/resources/EDIFACT-Common/IBM_EDI_Format.dfdl.xsd
@@ -69,7 +69,7 @@
                              textBooleanJustification="left"
                              sequenceKind="ordered"
                              choiceLengthKind="implicit"
-                             separator="" separatorPolicy="suppressed" separatorPosition="infix"
+                             separator="" separatorSuppressionPolicy="anyEmpty" separatorPosition="infix"
                              initiator=""
                              terminator=""
                              documentFinalTerminatorCanBeMissing="no"
@@ -131,17 +131,17 @@
 
             <dfdl:defineFormat name="EDISegmentSequenceFormat">
                 <dfdl:format separator="{$ibmEdiFmt:FieldSep}" separatorPosition="prefix"
-                             separatorPolicy="suppressedAtEndStrict" terminator=""/>
+                             separatorSuppressionPolicy="trailingEmpty" terminator=""/>
             </dfdl:defineFormat>
 
             <dfdl:defineFormat name="EDICompositeSequenceFormat">
                 <dfdl:format separator="{$ibmEdiFmt:CompositeSep}" separatorPosition="infix"
-                             separatorPolicy="suppressedAtEndLax" terminator=""/>
+                             separatorSuppressionPolicy="trailingEmptyStrict" terminator=""/>
             </dfdl:defineFormat>
 
             <dfdl:defineFormat name="EDISegmentPartRepeatSequenceFormat">
                 <dfdl:format separator="{$ibmEdiFmt:RepeatSep}" separatorPosition="infix"
-                             separatorPolicy="suppressedAtEndLax" terminator=""/>
+                             separatorSuppressionPolicy="trailingEmptyStrict" terminator=""/>
             </dfdl:defineFormat>
 
         </xsd:appinfo>

--- a/edg/pom.xml
+++ b/edg/pom.xml
@@ -16,7 +16,7 @@
         <dependency>
             <groupId>com.fasterxml.jackson.core</groupId>
             <artifactId>jackson-databind</artifactId>
-            <version>2.15.3</version>
+            <version>2.16.1</version>
         </dependency>
         <dependency>
             <groupId>com.github.spullara.mustache.java</groupId>
@@ -36,7 +36,7 @@
         <dependency>
             <groupId>org.apache.daffodil</groupId>
             <artifactId>daffodil-tdml-processor_2.12</artifactId>
-            <version>3.6.0</version>
+            <version>3.8.0</version>
             <scope>test</scope>
         </dependency>
         <dependency>

